packages/flame/lib/src/components/input/hud_button_component.dart

```diff
@@ -8,7 +9,7 @@ import 'package:flutter/rendering.dart' show EdgeInsets;
 /// Note: You have to set the [button] in [onLoad] if you are not passing it in
 /// through the constructor.
 class HudButtonComponent extends ButtonComponent
-    with HasGameRef, ComponentViewportMargin {
+    with HasGameRef<FlameGame>, ComponentViewportMargin {
```

Comment:
How come we are explicitly adding the generics here now?

Reply:
The generic is now based on `Game`, not on `FlameGame`.

Reply:
Added a comment about it here:
https://github.com/flame-engine/flame/pull/1828#discussion_r979249217

Reply:
Reverted this to how it was before